app/ui/browser-modern/sagas/page-sagas.js

```diff
@@ -77,6 +79,30 @@ export function* destroyPageSession({ id, withUI }) {
   }
 }
 
+function* forkPage(pageId, historyIndex) {
+  const currentPageIndex = yield select(PageSelectors.getPageIndexById, pageId);
+  const historyState = yield select(PageSelectors.getPageHistory, pageId);
+  const historyList = escape(JSON.stringify(historyState.toJS().map(history => history.url)));
+  const url = `tofino://sessionrestore/?history=${historyList}&historyIndex=${historyIndex}`;
```

Comment:
Define this url in endpoints.js, and use constants for addressing the tofino:// protocol.